### DISCUSSION TITLE
Changing Thread Dataset

### DISCUSSION THREAD

**Discussion by ssenthu** [openthread/openthread#9449] [Q&A] [answered]
Hi,

Once we started the Thread network with many sensors (after depolyment)
Is that possble to change any of the parameters using ot-ctl command so that all the router and sensor can follow without re-commisoning? 

Which dataset parameters are safe to change without affecting the network?
Ext PAN ID
PAN ID
Network Key
Network Name
Channel

**Comment by abtink** [comment, marked as answer]
You can check and use the [`dataset updater`](https://github.com/openthread/openthread/blob/93e72bcbf9b9a3a59a37b6db509851d85e5bcc2c/src/cli/cli_dataset.cpp#L1153-L1180) CLI and the related [`otDataseUpdater` OT APIs](https://github.com/openthread/openthread/blob/93e72bcbf9b9a3a59a37b6db509851d85e5bcc2c/include/openthread/dataset_updater.h). The `otDatasetUpdater` APIs can be used to update a subset (or all) fields in the Operational Dataset of the Thread mesh network (on all nodes in the network).

**Reply by jwhui** [reply]
Yes, take a look at `dataset updater`. I guess it would be helpful to provide some documentation in [src/cli/README_DATASET.md](https://github.com/openthread/openthread/blob/main/src/cli/README_DATASET.md), which I forgot to do in #6227.

The Dataset Updater component makes use of Thread's Pending Operational Dataset feature, which first distributes the Operational Dataset and schedules a time in the future for all devices to switch over. For parameters that affect connectivity (e.g. Network Key, PAN ID, Channel), there is a risk that some devices may not receive the update and get left behind. For example, if a device was disconnected during the time the Pending Operational Dataset was distributed and was made active.

**Reply by ssenthu** [reply]
Thanks a lot @abtink @jwhui for your quick answer. 
Can it be done with ot-ctl command?

for example, if i want to change the network name after 1000seconds, what is the procedure?

>ot-ctl dataset networkname ssss
>ot-ctl dataset delay 1000

>ot-ctl dataset active
Network Name: Ssss1234   <-- current

>ot-ctl dataset
Network Name: ssss           <--new, 

>dataset pending
Error 23: NotFound

**Reply by jwhui** [reply]
Yes, `ot-ctl` is a way to use OT CLI commands. You need to use `dataset updater start`. See the implementation here:
https://github.com/openthread/openthread/blob/93e72bcbf9b9a3a59a37b6db509851d85e5bcc2c/src/cli/cli_dataset.cpp#L1153-L1180

**Reply by ssenthu** [reply]
Thank a lot @jwhui . I tested and works fine.

**Comment by ssenthu** [comment]
[heart]         Sen Senthuran (CW) reacted to your message:
________________________________

**Comment by ssenthu** [comment]
[heart]         Sen Senthuran (CW) reacted to your message:
________________________________

**Comment by ssenthu** [comment]
[heart]         Sen Senthuran (CW) reacted to your message:
________________________________
